### PR TITLE
[14.0][FIX] fleet: remove obsolete table fks

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Let's greenify the repo.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This is not the solution, as the check is not at FK level. We should prevent that some data are removed marking them as noupdate=1 for example. Maybe this is something only of the current DB we are using, so the other alternative is to fix such DB.

**Review by legalsylvain — COMMENTED** [review]
Hi @MiquelRForgeFlow. Thanks a lot for trying to fix that annoying CI problem.

FYI, if you want to make test locally, the database used for this 13 -> 14 migration is available here : 

https://github.com/OCA/OpenUpgrade/releases/download/databases/13.0.psql

regards.

**Comment by MiquelRForgeFlow** [comment]
PR is green, ready to merge! :)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ jobs:
       PGHOST: "localhost"
       PGPASSWORD: "odoo"
       PGUSER: "odoo"
+      OPENUPGRADE_USE_DEMO: "ok"
```
Comment (legalsylvain) [inline .github/workflows/test.yml]: Minor remark / non blocking : value is not very classic. 1 or True could be better. Don't you think ?

File: openupgrade_framework/odoo_patch/odoo/modules/graph.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+
 import odoo
 from odoo.modules.graph import Graph
 
 
 def update_from_db(self, cr):
     """Prevent reloading of demo data from the new version on major upgrade"""
     Graph.update_from_db._original_method(self, cr)
+    if os.environ.get("OPENUPGRADE_USE_DEMO", "") == "ok":
+        return
```
Comment (legalsylvain) [inline openupgrade_framework/odoo_patch/odoo/modules/graph.py:13]: Not sure to understand. With that path, in witch case the following request (``UPDATE ir_module_module SET demo = false`` will be executed ? 
- in the CI with demo data - > no.
- in production without demo data -> no.

Maybe I missed something, and sorry if my question is not relevant.

**Review by legalsylvain — COMMENTED** [review]
Thanks a lot for your work @MiquelRForgeFlow.
Some questions : 

- (minor / non blocking) : as you introduce a new environment variable, could you add a few words in the description of the module ``openupgrade_framework``

- @StefanRijnhart : could you take a look on this one ?

File: openupgrade_framework/odoo_patch/odoo/modules/graph.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply openupgrade_framework/odoo_patch/odoo/modules/graph.py:13]: It will only executed in the case of production databases generated with demo data and then the demo data was customized. Thus, the script will transform that database to a not demo one.

File: .github/workflows/test.yml (hunk 1 above)
Comment (MiquelRForgeFlow) [reply .github/workflows/test.yml]: Not sure if it can be True. That's an yml file, not python file.

Comment (MiquelRForgeFlow) [reply .github/workflows/test.yml]: It seems we can only pass strings.

File: openupgrade_framework/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -17,6 +17,9 @@ OpenUpgrade migration scripts:
 * Run migration scripts for modules that are installed as new dependencies
   of upgraded modules (when there are such scripts for those particular
   modules)
+* Production databases generated with demo data, will be transformed to
+  non-demo ones. If you want to avoid that, you have pass in environment
+  the key OPENUPGRADE_USE_DEMO = "yes".
```
Comment (pedrobaeza) [inline openupgrade_framework/readme/DESCRIPTION.rst]:
```suggestion
  non-demo ones. If you want to avoid that, you have to pass through the
  environment variable OPENUPGRADE_USE_DEMO, the value "yes".
```

File: openupgrade_scripts/scripts/fleet/14.0.0.1/pre-migration.py [hunk 4]
```diff
@@ -9,3 +9,6 @@ def migrate(env, version):
         env.cr,
         {"fleet_vehicle_log_contract": [("state", None, None), ("name", None, None)]},
     )
+    openupgrade.remove_tables_fks(
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/fleet/14.0.0.1/pre-migration.py:12]: Why is this needed?

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/fleet/14.0.0.1/pre-migration.py:12]: because... obsolete tables, always better to remove the fks
